### PR TITLE
4: Magpie NPC

### PR DESCRIPTION
**Pull request by nreyana**
Closes https://github.com/UQcsse3200/2024-studio-2/issues/205.

### CODE REVIEW

**Review by EllaBerglas — COMMENTED** [review]
Just fix the one comment otherwise looks good

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java [hunk 1]
```diff
@@ -724,6 +728,113 @@ void TestSnakeSetPosition() {
...
+    /**
+<<<<<<< HEAD
+     * Tests Creation of a magpie.
+     */
+    @Test
+    void TestMagpieCreation() {
+        Assertions.assertNotNull(magpie, "magpie should not be null.");
+    }
+
+    /**
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java:740]: Most of these tests can probably be combined into a `testInititialisation` test - I don't think you need so many tests just to test the constructor.

**Review by mitsudz — APPROVED** [review]
Just pls refactor the tests and remove the comment Ella mentioned, but looks good other than that.

**Review by LucasRyall — APPROVED** [review]
As others have stated the tests need to be redone, and as I have mentioned in previous PR from your team, you shouldn't be re initialising every entity and service before every test, this test file already has a constructor and these should all be initialised once in the constructor.

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java [hunk 2]
```diff
@@ -724,6 +728,113 @@ void TestSnakeSetPosition() {
         assertEquals(pos, snake.getPosition());
     }
 
+    /**
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (nreyana) [reply source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java:732]: missed that by accident, thank you!

**Comment by BaileyS03** [comment]
> As others have stated the tests need to be redone, and as I have mentioned in previous PR from your team, you shouldn't be re initialising every entity and service before every test, this test file already has a constructor and these should all be initialised once in the constructor.

It isn't it's changed from BeforeEach to BeforeAll so it should only run once.

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java (hunk 1 above)
Comment (nreyana) [reply source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java]: all done!! thank you

**Pull request merged**